Parser/Parser.fs

```diff
@@ -618,19 +634,44 @@ module Parse =
                     <|>
                 (isProto3 >>. pStrMessageType_P3)
 
-            let pRpcOptions =
```

Comment:
Something about this section feels off.  I'm pretty sure it can still be defined in terms of a `pipe4`, which to me better communicates the form of the `TRPC` production.  Feels like there should be a common parser for all places where this type of `option` statement can occur.

Will try and look at this further when I get more time.